packages/contract/lib/utils/index.js

```diff
@@ -290,38 +268,32 @@ const Utils = {
   },
 
   // parses known contract instance networks
-  parseKnownNetworks(
+  async parseKnownNetworks(
     { networks, currentProvider, setNetwork, network_id },
     gasLimit
   ) {
-    // wrap uri matching in a promise to allow provider.send time to resolve
-    // (.send call happens in BlockchainUtils.matches)
-    return new Promise((accept, reject) => {
-      // go through all the networks that are listed as
-      // blockchain uris and see if they match
-      const uris = Object.keys(networks).filter(
-        network => network.indexOf("blockchain://") === 0
-      );
-      const matches = uris.map(uri =>
-        BlockchainUtils.matches.bind(BlockchainUtils, uri, currentProvider)
-      );
+    // go through all the networks that are listed as
+    // blockchain uris and see if they match
+    const uris = Object.keys(networks).filter(
+      network => network.indexOf("blockchain://") === 0
+    );
 
-      Utils.parallel(matches, (err, results) => {
-        if (err) reject(err);
+    const results = [];
+    for (const uri of uris) {
+      results.push(await BlockchainUtils.matches(uri, currentProvider));
```

Comment:
This can be parallelized.

Reply:
This I would argue we should just leave it for now until we need to optimize just for the sake of clarity. I'm not sure if we gain much speed-wise here. Do we?

Reply:
I don't see the purpose in this entire loop otherwise! `parseKnownNetworks` is an async function so we can just `Promise.all(results)` or really just combine these two loops.

Reply:
I don't see the purpose in this entire loop otherwise! parseKnownNetworks is an async function so we can just Promise.all(results) or really just combine these two loops.

Reply:
I don't really follow what you are saying. What do you mean you don't see the purpose otherwise?  The purpose is to perform the check. Whether or not it is done in parallel is a separate issue. Or am I misunderstanding something?

Reply:
It seems silly to loop through `uris` to collect an array of results to loop through again.  I think `await`ing defeats the purpose of looping through a first time, why not just combine these loops or collect an array of promises?

Reply:
Oh I see what you mean, let me take a look and see if it is clear and makes sense to loop just once.

Reply:
What do you think about that?